Tasks/AzureRmWebAppDeployment/azurermwebappdeployment.ts

```diff
@@ -1,141 +1,31 @@
-                }
-                else {
-                    await msDeploy.DeployUsingMSDeploy(webPackage, taskParams.WebAppName, msDeployPublishingProfile, taskParams.RemoveAdditionalFilesFlag,
-                    taskParams.ExcludeFilesFromAppDataFlag, taskParams.TakeAppOfflineFlag, taskParams.VirtualApplication, taskParams.SetParametersFile,
-                    taskParams.AdditionalArguments, isFolderBasedDeployment, taskParams.UseWebDeploy);
-                }
-            }
-            else {
-                tl.debug("Initiated deployment via kudu service for webapp package : ");
-                await kuduServiceUtility.deployWebPackage(webPackage, physicalPath, taskParams.VirtualApplication, taskParams.TakeAppOfflineFlag);
-            }
-        }
-
-        if(!taskParams.isContainerWebApp) {
-            if(taskParams.AppSettings) {
-                var customApplicationSettings = ParameterParser.parse(taskParams.AppSettings);
-                await appServiceUtility.updateAndMonitorAppSettings(customApplicationSettings);
-            }
-
-            if(taskParams.ConfigurationSettings) {
-                var customApplicationSettings = ParameterParser.parse(taskParams.ConfigurationSettings);
-                await appServiceUtility.updateConfigurationSettings(customApplicationSettings);
-            }
-        }
-        else {
-            tl.debug('App Settings and config settings are already updated during container based deployment.')
-        }
+        var deploymentProvider = DeploymentFactory.GetDeploymentProvider(taskParams);
 
-        if(taskParams.ScriptType) {
-            await kuduServiceUtility.runPostDeploymentScript(taskParams, virtualApplicationPath);
-        }
+        tl.debug("Predeployment Step Started");
+        await deploymentProvider.PredeploymentStep();
 
-        await appServiceUtility.updateScmTypeAndConfigurationDetails();
+        tl.debug("Deployment Step Started");
+        await deploymentProvider.DeployWebAppStep();
     }
     catch(error) {
+        tl.debug("Deployment Failed with Error: " + error);
         isDeploymentSuccess = false;
         tl.setResult(tl.TaskResult.Failed, error);
     }
     finally {
-        if(kuduServiceUtility) {
-            await addReleaseAnnotation(azureEndpoint, appService, isDeploymentSuccess);
-            let activeDeploymentID: string = await kuduServiceUtility.updateDeploymentStatus(isDeploymentSuccess, null, {'type': 'Deployment', slotName: appService.getSlot()});
-            if(zipDeploymentID && activeDeploymentID && isDeploymentSuccess) {
-                await kuduServiceUtility.postZipDeployOperation(zipDeploymentID, activeDeploymentID);
-            }
-        }
-        else {
-            tl.debug('Cannot update deployment status as Kudu is not initialized');
-        }
+        await deploymentProvider.UpdateDeploymentStatus(isDeploymentSuccess);
+        tl.debug(isDeploymentSuccess ? "Deployment Succeded" : "");
```

Comment:
Should we log deployment failed when isDeploymentSuccess is false

Reply:
The error logs are getting printed in catch block above.